README.md

```diff
@@ -17,75 +17,81 @@ This is a general purpose library for working with Content Security Policy polic
 mvn install
 ```
 
+### A Note on CSP
+
+The CSP specification is fairly complex even if you only care about the latest version. However, in practice you are likely to care that your policy does the things you intend it to on the browsers you care about, which are likely to implement different and potentially broken subsets of the specification (and in some cases supersets). And there are inevitable tradeoffs to be made regarding the size of your policy vs the security it provides.
+
+As such, this project does not attempt to provide a one-size-fits-all way to manipulate a policy purely in terms of its effects - the full set of effects across all browsers is too vast to provide an effective API in general. It can help you build up a policy based on the directives and source-expressions you want, but to ensure your policy is correct, for your own definition of correct, there is no alternative to testing it on the real browsers you care about.
+
+
 ### Create a Policy
 
-Parse a policy using one of the `Parser.parse` static methods. An `Origin` or `String` may be given as the origin. The third parameter, if given, will be populated with notices.
+Parse a policy using either `Policy.parseSerializedCSP` or `Policy.parseSerializedCSPList`. The second parameter will be called for each warning or error.
 
 ```java
-ArrayList<Notice> notices = new ArrayList<>();
-Origin origin = URI.parse("http://example.com");
-String policyText = "...";
-Policy p = Parser.parse(policyText, origin, notices);
+String policyText = "script-src 'none'";
+Policy policy = Policy.parseSerializedCSP("default-src 'none'", (severity, message, directiveIndex, valueIndex) -> {
```

Comment:
Assigned `policyText` on 32 but then didn't use it in 33?

Reply:
Nice catch, fixed.